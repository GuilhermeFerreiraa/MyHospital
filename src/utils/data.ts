
const JOBS = [
	{
		"id": "1",
		"title": "Hospital East Jacklynland",
		"job_quantity": 3,
		"description": "Hospital Clínico localizado na região North de Rachelleworth.",
		"cover": "https://images.unsplash.com/photo-1512102438733-bfa4ed29aef7?q=80&w=1374&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1512102438733-bfa4ed29aef7?q=80&w=1374&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "f0a6f623-35d1-4843-9b30-b1558744343a",
				"created_at": "2024-07-17T13:43:02.523Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2024-11-20T15:30:41.703Z",
					"ends_at": "2024-11-20T23:30:41.703Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "150a924b-b070-43c9-97b9-c8618fbb09c1",
				"created_at": "2024-07-16T20:50:35.903Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2024-08-28T08:40:23.275Z",
					"ends_at": "2024-08-28T20:40:23.275Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "95988ddf-85a2-47dc-9f1d-dc67b9809114",
				"created_at": "2024-07-17T05:10:36.711Z",
				"role": "Enfermeiro",
				"requirement": "Experiência com cuidados intensivos é um diferencial.",
				"schedule": {
					"starts_at": "2025-03-21T09:33:39.505Z",
					"ends_at": "2025-03-21T21:33:39.505Z"
				},
				"shift_type": "Noturno"
			}
		]
	},

	{
		"id": "2",
		"title": "Hospital Fort Mae",
		"job_quantity": 3,
		"description": "Hospital Clínico localizado na região Northeast de Milwaukee.",
		"cover": "https://plus.unsplash.com/premium_photo-1675686363477-c28d5bf65adb?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://plus.unsplash.com/premium_photo-1675686363477-c28d5bf65adb?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "17752b5a-5969-4785-a6e1-2fdd49fe44b8",
				"created_at": "2024-07-17T06:38:12.144Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2024-09-12T01:12:18.070Z",
					"ends_at": "2024-09-12T13:12:18.070Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "9facf76f-663d-435c-8bee-cf2dd8a8ca41",
				"created_at": "2024-07-17T12:22:27.131Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-06-21T09:23:13.491Z",
					"ends_at": "2025-06-21T21:23:13.491Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "14ab267f-a960-4c8f-ab82-5fabfa9ee8e8",
				"created_at": "2024-07-17T08:26:26.530Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-12-20T06:12:14.529Z",
					"ends_at": "2024-12-20T18:12:14.529Z"
				},
				"shift_type": "Matinal"
			}
		]
	},

	{
		"id": "3",
		"title": "Hospital Dwightborough",
		"job_quantity": 3,
		"description": "Hospital Clínico localizado na região Northeast de North Faustinoberg.",
		"cover": "https://images.unsplash.com/photo-1605725961202-18c2678ea41e?q=80&w=1471&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1605725961202-18c2678ea41e?q=80&w=1471&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "11665c8a-539d-4ff1-b50e-a692f06a9b95",
				"created_at": "2024-07-17T18:18:32.513Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-08-21T18:54:59.415Z",
					"ends_at": "2024-08-22T06:54:59.415Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "0591343f-0f39-4e61-b1d4-78b5711bc598",
				"created_at": "2024-07-17T07:32:45.517Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2024-09-19T02:55:41.523Z",
					"ends_at": "2024-09-19T14:55:41.523Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "15716024-0b21-445b-9753-f43a895cbfb8",
				"created_at": "2024-07-16T22:29:22.860Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-06-25T18:27:58.280Z",
					"ends_at": "2025-06-26T06:27:58.280Z"
				},
				"shift_type": "Noturno"
			}
		]
	},

	{
		"id": "4",
		"title": "Hospital South Asiafield",
		"job_quantity": 5,
		"description": "Hospital Clínico localizado na região North de Fort Shayna.",
		"cover": "https://images.unsplash.com/photo-1580281780460-82d277b0e3f8?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1580281780460-82d277b0e3f8?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "a78fdcf8-aba0-4156-b3d2-e6dfbd55d6b2",
				"created_at": "2024-07-17T17:31:15.948Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2025-05-25T08:47:51.466Z",
					"ends_at": "2025-05-25T20:47:51.466Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "a8fea830-dabd-45d3-9b9d-bf8447d25e43",
				"created_at": "2024-07-17T13:43:39.878Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2025-04-16T05:42:21.093Z",
					"ends_at": "2025-04-16T17:42:21.093Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "6d02c4f4-bff1-4412-9ed2-468a8eab4b25",
				"created_at": "2024-07-17T17:24:15.938Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-10-27T03:06:32.373Z",
					"ends_at": "2024-10-27T15:06:32.373Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "757daedb-fc41-40df-87e5-a249a996b108",
				"created_at": "2024-07-17T01:56:03.831Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-06-09T14:32:26.632Z",
					"ends_at": "2025-06-10T02:32:26.632Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "33ba8fe6-baa5-4c79-b5b7-70b715d401e2",
				"created_at": "2024-07-17T03:30:47.459Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2024-12-17T09:43:00.059Z",
					"ends_at": "2024-12-17T21:43:00.059Z"
				},
				"shift_type": "Matinal"
			}
		]
	},
	{
		"id": "5",
		"title": "Hospital Lake Ayden",
		"job_quantity": 3,
		"description": "Hospital Clínico localizado na região Northwest de East Sammie.",
		"cover": "https://plus.unsplash.com/premium_photo-1661894592852-5cfb513eefb4?q=80&w=1287&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://plus.unsplash.com/premium_photo-1661894592852-5cfb513eefb4?q=80&w=1287&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "8c7efb89-06f0-47fe-8b38-16c67f043e2c",
				"created_at": "2024-07-17T02:19:02.237Z",
				"role": "Enfermeiro",
				"requirement": "Experiência com cuidados intensivos é um diferencial.",
				"schedule": {
					"starts_at": "2025-07-11T15:13:02.539Z",
					"ends_at": "2025-07-12T03:13:02.539Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "3927fe74-6e89-46d2-aca2-da1499d0837c",
				"created_at": "2024-07-17T11:27:39.940Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2024-07-24T04:13:32.697Z",
					"ends_at": "2024-07-24T16:13:32.697Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "6cc9f084-1396-4167-bdff-59455ac62f56",
				"created_at": "2024-07-17T13:24:27.671Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência com cuidados intensivos é um diferencial.",
				"schedule": {
					"starts_at": "2025-07-12T04:42:56.477Z",
					"ends_at": "2025-07-12T16:42:56.477Z"
				},
				"shift_type": "Matinal"
			}
		]
	},

	{
		"id": "6",
		"title": "Hospital Hiltonfort",
		"job_quantity": 4,
		"description": "Hospital Clínico localizado na região East de Makaylafield.",
		"cover": "https://plus.unsplash.com/premium_photo-1661963558819-5d4201ae5dce?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://plus.unsplash.com/premium_photo-1661963558819-5d4201ae5dce?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "a9d6affb-edef-45ea-a578-5bf4d3f5f856",
				"created_at": "2024-07-16T22:06:34.086Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência com cuidados intensivos é um diferencial.",
				"schedule": {
					"starts_at": "2024-12-12T00:46:52.908Z",
					"ends_at": "2024-12-12T12:46:52.908Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "f23657c8-fa23-4077-93cb-41c0ab3569a1",
				"created_at": "2024-07-17T10:26:42.461Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-07-21T01:07:55.715Z",
					"ends_at": "2024-07-21T13:07:55.715Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "1791f188-693c-436f-993a-5da23c598b08",
				"created_at": "2024-07-17T13:15:50.296Z",
				"role": "Enfermeiro",
				"requirement": "Experiência com cuidados intensivos é um diferencial.",
				"schedule": {
					"starts_at": "2024-12-29T20:10:41.705Z",
					"ends_at": "2024-12-30T08:10:41.705Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "fdb298a4-9ddb-42df-86ec-fc1641f68dc7",
				"created_at": "2024-07-16T22:07:33.270Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2025-04-18T19:53:19.302Z",
					"ends_at": "2025-04-19T07:53:19.302Z"
				},
				"shift_type": "Noturno"
			}
		]
	},
	{
		"id": "7",
		"title": "Hospital Lake Jermainstead",
		"job_quantity": 5,
		"description": "Hospital Clínico localizado na região South de New Katherineville.",
		"cover": "https://images.unsplash.com/photo-1626315869436-d6781ba69d6e?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1626315869436-d6781ba69d6e?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "87df2915-16d1-40c3-849f-e78b8778b573",
				"created_at": "2024-07-17T14:49:12.800Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2024-11-07T08:01:23.813Z",
					"ends_at": "2024-11-07T20:01:23.813Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "1d11265b-9372-4461-b830-a39c0a92fd61",
				"created_at": "2024-07-17T13:09:22.726Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-12-17T04:44:16.759Z",
					"ends_at": "2024-12-17T16:44:16.759Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "c36ba560-7b22-4f1a-b181-0d23fe0078b5",
				"created_at": "2024-07-17T09:19:09.072Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-08-05T11:37:54.070Z",
					"ends_at": "2024-08-05T23:37:54.070Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "615a8ff4-2746-4d06-bd8a-911a1aeec3e6",
				"created_at": "2024-07-17T05:21:41.398Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2024-08-05T07:14:40.809Z",
					"ends_at": "2024-08-05T19:14:40.809Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "3fb33e3c-89da-43d3-b066-ddcb029ce226",
				"created_at": "2024-07-17T13:36:16.483Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2025-04-15T22:51:29.179Z",
					"ends_at": "2025-04-16T10:51:29.179Z"
				},
				"shift_type": "Matinal"
			}
		]
	},
	{
		"id": "8",
		"title": "Hospital Jonesview",
		"job_quantity": 4,
		"description": "Hospital Clínico localizado na região West de Ciaraside.",
		"cover": "https://images.unsplash.com/photo-1549560826-4b7bfe23f37b?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1549560826-4b7bfe23f37b?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "9090b463-b7b1-4b16-a989-bab4b173f0dd",
				"created_at": "2024-07-16T20:41:19.840Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2025-03-04T03:15:48.908Z",
					"ends_at": "2025-03-04T15:15:48.908Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "3b2a7621-b1f6-480c-9760-6c80aca835d4",
				"created_at": "2024-07-17T06:41:56.087Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2025-05-24T18:18:54.365Z",
					"ends_at": "2025-05-25T06:18:54.365Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "1dcb66e9-db42-4799-84b2-0820db13f10f",
				"created_at": "2024-07-16T20:30:24.833Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2024-08-29T11:58:11.520Z",
					"ends_at": "2024-08-29T23:58:11.520Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "ae8da4c7-e935-4cc5-a215-8d67c9f09808",
				"created_at": "2024-07-17T12:04:47.289Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-10-23T06:13:13.950Z",
					"ends_at": "2024-10-23T18:13:13.950Z"
				},
				"shift_type": "Matinal"
			}
		]
	},
	{
		"id": "9",
		"title": "Hospital North Bernie",
		"job_quantity": 4,
		"description": "Hospital Clínico localizado na região North de Port Guiseppetown.",
		"cover": "https://images.unsplash.com/photo-1538108149393-fbbd81895907?q=80&w=1528&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1538108149393-fbbd81895907?q=80&w=1528&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "57acbbef-4994-45e5-97a1-6d67967b4a42",
				"created_at": "2024-07-17T08:14:51.409Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2025-04-24T20:52:13.291Z",
					"ends_at": "2025-04-25T08:52:13.291Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "6058a282-3202-4373-b641-073d151c19d5",
				"created_at": "2024-07-17T13:23:09.162Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-10-19T11:52:28.095Z",
					"ends_at": "2024-10-20T01:52:28.095Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "516043de-49ac-4acd-9e93-62d02991c7db",
				"created_at": "2024-07-17T14:56:58.989Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-07-29T15:42:09.586Z",
					"ends_at": "2024-07-30T03:42:09.586Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "4ae2f08e-662c-4462-a658-13fc3d900178",
				"created_at": "2024-07-17T18:48:38.641Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2024-11-03T12:43:54.359Z",
					"ends_at": "2024-11-04T00:43:54.359Z"
				},
				"shift_type": "Matinal"
			}
		]
	},
	{
		"id": "10",
		"title": "Hospital Fort Jettborough",
		"job_quantity": 5,
		"description": "Hospital Clínico localizado na região Southwest de Lake Terrence.",
		"cover": "https://images.unsplash.com/photo-1512678080530-7760d81faba6?q=80&w=1474&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"thumbnail": "https://images.unsplash.com/photo-1512678080530-7760d81faba6?q=80&w=1474&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
		"shifts": [
			{
				"id": "963d19c0-4e94-4c9f-935c-e185763e78dc",
				"created_at": "2024-07-17T07:35:59.357Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-07-09T01:29:08.012Z",
					"ends_at": "2025-07-09T13:29:08.012Z"
				},
				"shift_type": "Noturno"
			},
			{
				"id": "2f397a35-c39b-46d9-9bbf-8f4c50e19d01",
				"created_at": "2024-07-17T09:40:54.919Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 1 ano em clínicas.",
				"schedule": {
					"starts_at": "2024-08-27T06:55:20.761Z",
					"ends_at": "2024-08-27T18:55:20.761Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "692b586e-469f-466d-b3a7-5e0c420922b2",
				"created_at": "2024-07-17T12:54:58.100Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 2 anos no ramo da saúde.",
				"schedule": {
					"starts_at": "2025-07-15T19:13:18.004Z",
					"ends_at": "2025-07-16T07:13:18.004Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "cf8416db-0f25-4f0a-b4dd-6c3af15efc2d",
				"created_at": "2024-07-17T19:07:35.860Z",
				"role": "Técnico de Enfermagem",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-02-02T06:06:08.069Z",
					"ends_at": "2025-02-02T18:06:08.069Z"
				},
				"shift_type": "Matinal"
			},
			{
				"id": "f22ebf63-c67e-4727-b851-df7ca25006fd",
				"created_at": "2024-07-17T00:49:00.657Z",
				"role": "Enfermeiro",
				"requirement": "Experiência de no mínimo 3 anos em hospitais.",
				"schedule": {
					"starts_at": "2025-07-04T09:54:58.080Z",
					"ends_at": "2025-07-04T21:54:58.080Z"
				},
				"shift_type": "Matinal"
			}
		]
	}
]

const JobOffer = JOBS.map((item) => item).flat();

type JobOfferProps = (typeof JobOffer)[0];

export { JOBS, JobOffer, JobOfferProps };
