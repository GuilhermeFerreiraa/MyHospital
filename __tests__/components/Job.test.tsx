import React from 'react';
import renderer from 'react-test-renderer';
import Job from '@/components/job';
import { JobOfferProps } from '@/utils/data';

const mockData: JobOfferProps = {
	title: "Hospital Fort Jettborough",
	description: "Hospital Clínico localizado na região Southwest de Lake Terrence.",
	thumbnail: 'https://example.com/thumbnail.jpg',
	job_quantity: 5,
	id: '1',
	cover: 'https://example.com/thumbnail.jpg',
	shifts: []
};

describe('<Job />', () => {
	it('deve renderizar corretamente com as props fornecidas', () => {
		const tree = renderer.create(
			<Job
				data={mockData}
				showCandidaciesOnly={false}
				onPress={() => { }}
			/>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('deve ter os componentes corretos e seus textos', () => {
		const component = renderer.create(
			<Job
				data={mockData}
				showCandidaciesOnly={false}
				onPress={() => { }}
			/>
		);

		const instance = component.root;

		expect(instance.findByProps({ testID: 'job-component' })).toBeDefined();

		expect(instance.findByProps({ testID: 'job-thumbnail' }).props.source.uri).toBe(mockData.thumbnail);

		expect(instance.findByProps({ testID: 'job-title' }).props.children).toBe(mockData.title);

		expect(instance.findByProps({ testID: 'job-description' }).props.children).toBe(mockData.description);

		const numberRoleText = instance.findByProps({ testID: 'job-number-role' }).props.children;
		expect(Array.isArray(numberRoleText) ? numberRoleText.join('') : numberRoleText).toBe(`${mockData.job_quantity}x vagas disponíveis`);
	});
});
