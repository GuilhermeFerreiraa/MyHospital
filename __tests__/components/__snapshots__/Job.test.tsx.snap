// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Job /> deve renderizar corretamente com as props fornecidas 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "row",
      "marginBottom": 8,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "width": "100%",
    }
  }
  testID="job-component"
>
  <Image
    source={
      {
        "uri": "https://example.com/thumbnail.jpg",
      }
    }
    style={
      [
        {
          "width": 80,
        },
        {
          "height": 80,
        },
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
        },
      ]
    }
    testID="job-thumbnail"
  />
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginLeft": 12,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#0f172a",
            },
            {
              "width": "100%",
            },
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "fontFamily": "Inter_500Medium",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
          ]
        }
        testID="job-title"
      >
        Hospital Fort Jettborough
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#1e293b",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "lineHeight": 20,
          },
          {
            "marginTop": 2,
          },
        ]
      }
      testID="job-description"
    >
      Hospital Clínico localizado na região Southwest de Lake Terrence.
    </Text>
    <Text
      style={
        [
          {
            "color": "#23b0ba",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "lineHeight": 20,
          },
          {
            "marginTop": 2,
          },
        ]
      }
      testID="job-number-role"
    >
      5x vagas disponíveis
    </Text>
  </View>
</View>
`;
